shoop/xtheme/views/extra.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+import six
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.core.signals import setting_changed
+from django.http.response import HttpResponseNotFound
+
+from shoop.utils.importing import load
+from shoop.xtheme.theme import get_current_theme
+
+_VIEW_CACHE = {}
+
+
+def clear_view_cache(**kwargs):
+    _VIEW_CACHE.clear()
+
+
+setting_changed.connect(clear_view_cache, dispatch_uid="shoop.xtheme.views.extra.clear_view_cache")
+
+
+def _get_view_by_name(theme, view_name):
+    view = getattr(theme, "views", {}).get(view_name)
+    if isinstance(view, six.string_types):  # Handle loading views from specs
+        try:
+            view = load(view)
+        except ImproperlyConfigured:
+            if settings.DEBUG:
+                raise
+            return None
```

Comment:
* It's making the `_get_view_by_name` more complicated than it could be.
* It's hiding dependencies from all code analysis / refactoring tools.
* No autocomplete by IDEs. (as said in the django link I provided earlier)

Yeah, declarative is cool, but since it already is in a *.py file, why not take advantage of the full power of the Python language rather than parsing a class content like it was some sort of XML or something?

You could btw. also provide a Theme base class that does make things easier for dummies:
```python
class XmlDeclaredTheme(Theme):
    def get_views(self):
        return {
            spec: self._load_view_by_spec(spec)
            for (view_name, spec) in self._get_views_from_xml_file()
        }
```
and then just instruct the dummies to dump their view stuff to the XML file.